### PR TITLE
Add endpoint to log filtered locks on Lockserver

### PR DESCRIPTION
**Pull request by jboreiko**
repo port from #1448

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1633)

### CODE REVIEW

File: lock-api/src/main/java/com/palantir/lock/HeldLocksGrant.java [hunk 1]
```diff
@@ -106,7 +106,8 @@ public long getExpirationDateMs() {
      * Returns the set of locks which were successfully acquired, as a mapping
      * from descriptor to lock mode.
      */
-    public SortedLockCollection<LockDescriptor> getLocks() {
+    @Override
+    public SortedLockCollection<LockDescriptor> getLockDescriptors() {
```
Comment (gsheasby) [inline lock-api/src/main/java/com/palantir/lock/HeldLocksGrant.java]: Shouldn't we keep `getLocks` around as it's a public API method? We could just forward it to `getLockDescriptors` (and possibly deprecate it).

**Comment by gsheasby** [comment]
@jeremyk-91 why is this blocked? Looks like we just need to fix the API break.

**Comment by jeremyk-91** [comment]
Hmm, I don't remember exactly why this was marked as blocked during the triage meeting (probably pending on @datval or @jboreiko to address the API break?)

I had a quick look through the change and it looks fine apart from the API break, as mentioned.

**Comment by datval** [comment]
I think it blocked on me. Glenn requested to fix API brake and to do some refactoring around logging (aka to introduce separate logger class).

**Comment by gsheasby** [comment]
Yep, there were a few comments on #1448 about making the code look a bit nicer - those would be "nice to have" if you have the bandwidth this week, but not essential.

**Comment by rhero** [comment]
Hey @datval - just wanted to ping this and see if you think you'll be able to get to it this week.

**Comment by datval** [comment]
@rhero I don't think this will be merged during this week. Is it a problem? We are testing some different approach here on 0.4.x first

**Comment by rhero** [comment]
No, no problem at all! Just wanted to start being more diligent about checking on open PRs!

**Comment by rhero** [comment]
@datval pinging this PR again. Do you think you'll be able to take a look at it this week?

**Comment by datval** [comment]
There is PR on 0.4.x internally, when I have enough feedback, will push something here as well. We can close this meanwhile.

**Comment by rhero** [comment]
Sounds good, closing for now, but feel free to put another PR up and assign it to someone in London.

**Pull request closed without merging**